lib/wpc_proto/internal_modules/proto_config.c

```diff
@@ -276,7 +276,253 @@ bool Proto_config_init(void)
 app_proto_res_e Proto_config_handle_set_config(wp_SetConfigReq *req,
                                                wp_SetConfigResp *resp)
 {
-    return APP_RES_PROTO_NOT_IMPLEMENTED;
+    app_res_e res;
+    app_res_e global_res = APP_RES_OK;
+    bool config_has_changed = false;
+
+    // TODO: Add some sanity checks
+
+    // check any config change and apply them
+    wp_SinkNewConfig * cfg = &req->config;
+
+    //cfg->sink_id[16]; not used
+
+    if (cfg->has_node_role)
+    {
+        app_role_t options = 0;
+        if (cfg->node_role.flags_count > 0)
+        {
+            options |= (cfg->node_role.flags[0] == wp_NodeRole_RoleFlags_LOW_LATENCY)
```

Comment:
Small function for that? It is duplicated :-) And technically we could add new options later on (probably never :-) ) but roles_count may be 3.

Reply:
I started to use small fonction for get config, so I did the same here for get config (the other direction). But why duplicated, each one is called only once, no ?
If we want to support flag APP_ROLE_OPTION_RELAY, we need to add wp_NodeRole_RoleFlags_RELAY (but it's deprecated). As there is no formula to convert flags, is we want to support 3 flags, we need to change the code.
I changed the static check to remain compatible with 3 flags, and just alert if we have room for one flag.